cirq/contrib/noise_models/noise_models.py

```diff
@@ -142,3 +144,51 @@ def noisy_moment(self, moment: 'cirq.Moment',
                 moment,
                 ops.Moment(self.qubit_noise_gate(q) for q in system_qubits)
             ]
+
+
+class PerQubitDepolarizingNoiseModel(devices.NoiseModel):
+    """DepolarizingNoiseModel which allows depolarization probabilities to be
+    specified separately for each qubit.
+
+    Similar to depol_prob in DepolarizingNoiseModel, depol_prob_map should map
+    Qids in the device to their depolarization probability.
+    """
+
+    def __init__(
+            self,
+            depol_prob_map: Dict['cirq.Qid', float],
+    ):
+        """A depolarizing noise model with variable per-qubit noise.
+
+        Args:
+            depol_prob_map: Map of depolarizing probabilities for each qubit.
+        """
+        for qubit, depol_prob in depol_prob_map.items():
+            value.validate_probability(depol_prob, f'depol prob of {qubit}')
+        self.depol_prob_map = depol_prob_map
+
+    def noisy_moment(self, moment: 'cirq.Moment',
+                     system_qubits: Sequence['cirq.Qid']):
+        if _homogeneous_moment_is_measurements(moment):
+            return moment
+        else:
+            return [
+                moment,
+                ops.Moment(
+                    ops.DepolarizingChannel(self.depol_prob_map[q])(q)
```

Comment:
Since you are modeling the error based on 1-q error rates, I think you would want a depolarizing channel only for qubits in the moment, (exclude idle qubits?)

Reply:
Done. Left idle qubits unchanged for now - should they have an AmplitudeDamping channel based on the calibration T1 values?

Reply:
In practice, we generally treat the "idle gate" has having roughly the same fidelity as other single-qubit operations

Reply:
Got it. I'll leave more complex noise models for a follow-up PR.